File: angular-14-client/src/app/alunoturmacurso/model/alunoturmacurso.ts
import { Aluno } from 'src/app/alunos/model/alunos';
import { TurmaCursos } from 'src/app/turmacursos/model/turmacursos';

export class AlunoTurmaCurso {
   id: number;
   aluno : Aluno;
   nome: string;
   lstTurmaCursos : Array<TurmaCursos>;
}

File: spring-boot-server/src/main/java/com/sganzerla/model/AlunoTurmaCurso.java
package com.sganzerla.model;

import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
@Data
public class AlunoTurmaCurso {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;

    private Long aluno;

    private String nome;

    private Long turmaCurso;

}
